### ISSUE TITLE
New Invoice: Checkbox "Email Copy of Invoice" is useless

### ISSUE THREAD

**Issue by TerrapinSoftware** [magento/magento2#28511]
### Preconditions (*)

1. 2.3.5-p1
2. Magento 2.4-develop

### Steps to reproduce (*)

1. Go to Admin -> Stores -> Configuration -> Sales -> Sales Emails -> Invoice:
 set **Enable** to **Yes**;
set **Send Invoice Email Copy To** to **test@example.com**
2. Create Order;
3. Go to Admin -> Sales -> Orders and open created before order for edit;
4. Click on the **Invoice** button;
2. Uncheck "Email Copy of Invoice";
3. Generate Invoice;

### Expected result (*)

1. Email should not be sent to anyone

### Actual result (*)

1. Email sent both to Customer and Copy To

The behavior changed from M1 to M2. Regardless if you check that checkbox or not, the email is sent to both the customer and the Copy To address. If you disable the sending of invoices in the Config section, there is no way to send invoice emails at all, because the "Send Email" button in the View Invoice screen does nothing, and the checkbox is not displayed in the New Invoice section.

Looking at the source code, I find this in `vendor/magento/module-sales/Controller/Adminhtml/Order/Invoice/Save.php`, line 216:

```php
if (!empty($data['send_email']) || $this->salesData->canSendNewInvoiceEmail()) {
    $this->invoiceSender->send($invoice);
````
Looks good so far; if the config is "off", the checkbox would override the setting.

Unfortunately, the checkbox is not displayed at all when the config is "off".

Even if the checkbox would be visible, emails would not be sent, because in the file `vendor/magento/module-sales/Model/Order/Email/Sender/InvoiceSender.php` at line 101, I find this:

```php
public function send(Invoice $invoice, $forceSyncMode = false)
{
    $invoice->setSendEmail($this->identityContainer->isEnabled());
```
This picks up the "off" value from the config, thus overriding any checkbox value again.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [*] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by TerrapinSoftware** [comment]
OK. I see that this bug is very, very difficult to fix. Here is the fix.

In file `vendor/magento/module-sales/Controller/Adminhtml/Order/Invoice/Save.php`, line 216, replace

```php
if (!empty($data['send_email']) || $this->salesData->canSendNewInvoiceEmail()) {
    $this->invoiceSender->send($invoice);
```

with

```php
if (!empty($data['send_email']) && $this->salesData->canSendNewInvoiceEmail()) {
    $this->invoiceSender->send($invoice);
```

Change `||` to `&&`.

**Comment by kamal02mittal** [comment]
Yes, I have also applied the same 
replaced below
```
if (!empty($data['send_email']) || $this->salesData->canSendNewInvoiceEmail()) {
    $this->invoiceSender->send($invoice);
```

With 

```
if (!empty($data['send_email']) && $this->salesData->canSendNewInvoiceEmail()) {
    $this->invoiceSender->send($invoice);
```

OR

if (!empty($data['send_email'])) {
    $this->invoiceSender->send($invoice);

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-39026` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by aleron75** [comment]
I have a doubt about the expectation

>Email sent to "Send Invoice Email Copy To" (settings) only

I think _"Send Invoice Email Copy To"_ doesn't mean _"send an invoice by email only to the recipient configured in  `Send Invoice Email Copy To` field"_ but _"send a copy of the invoice to the customer"_ and thus, if configured, to the above-mentioned recipient. 

In other words, if we ask Magento to send an email to the customer, then it possibly can be sent as a copy to the configured additional recipient. 

But if we **don't** ask Magento to **send** the email (leaving the checkbox unchecked) we can't expect **any sent email**.

That said, it's likely that there is a bug related to sending/not sending the email, I have to go further on this but before that, I want to be sure what the expected behavior is.

@sidolov @engcom-Alfa what do you think about it?

**Comment by engcom-Alfa** [comment]
Hello @aleron75.

Yes, you are right. The expected result is incorrect. If we don't ask Magento to send the email we can't expect any sent email.
Let's imagine a situation when the admin user has limited rights and the configuration cannot be changed.
And for one of a thousand orders, you will not need to send letters. And such an administrator cannot do it in any way, copies will be sent anyway.
Even if the administrator has the rights, you need to delete the list of emails for sending copies in the configuration.

Therefore, the expected result should be that if we clear the "Invoice copy" checkbox, the emails will not be sent to anyone.

**Comment by magento-engcom-team** [comment]
Hi @TerrapinSoftware. Thank you for your report. 
The issue has been fixed in magento/magento2#30868 by @gwharton in 2.4-develop branch
Related commit(s):
 - [e0d8149083b51604b82dbcee3d412f5424e7aa60](https://github.com/magento/magento2/commit/e0d8149083b51604b82dbcee3d412f5424e7aa60)

The fix will be available with the upcoming 2.4.3 release.
